standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/CheckResult.java

```diff
@@ -95,22 +103,27 @@ public Set<PartitionResult> getPartitionsNotInMs() {
    * @param partitionsNotInMs
    *          a list of partitions not found in the metastore
    */
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP2", justification = "intended_to_do")
   public void setPartitionsNotInMs(Set<PartitionResult> partitionsNotInMs) {
     this.partitionsNotInMs = partitionsNotInMs;
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "intended_to_do")
   public Set<PartitionResult> getExpiredPartitions() {
     return expiredPartitions;
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP2", justification = "intended_to_do")
   public void setExpiredPartitions(final Set<PartitionResult> expiredPartitions) {
     this.expiredPartitions = expiredPartitions;
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "intended_to_do")
```

Comment:
Why are we suppressing warnings?

Reply:
We had upgraded spotbugs-maven-plugin from 4.0.0 to 4.5.0.0 for compatibility with JDK17. 
Each upgrade of this plugin enhances the rules for finding the potential bugs and a major enhancement was in 4.3.0 release where the change was `MS_EXPOSE_REP and EI_EXPOSE_REP are now reported for code returning a reference to a mutable object indirectly (e.g. via a local variable)`
Due to this change, a lot of new bugs arised in our codebase and since there is a complete different initiative to fix these bugs as part of [HIVE-23613](https://issues.apache.org/jira/browse/HIVE-23613) , for jdk17 migration we have suppressed these new warnings with intended_to_do token.

Reply:
nit: If you could add the jira number in the justification. That would be nice to have.

Reply:
Yes definitely, have committed the change to add HIVE Jira ( [HIVE-23613](https://issues.apache.org/jira/browse/HIVE-23613) ) in the justification of all newly added SuppressFBWarnings tags.